### ISSUE TITLE
Unable to read references with npm link

### ISSUE THREAD

**Issue by Ciantic** [TypeStrong/atom-typescript#330]
I'm using npm link for local package in Windows 8.1. Atom typescript marks the the `<reference />` and `import` as error not found. 

TSC works normally.

**Comment by basarat** [comment]
Sample please :heart:

**Comment by Ciantic** [comment]
```
cd mylocalpackage
npm link
cd ..
cd otherpackage
npm link mylocalpackage
```

`otherpackage\node_modules\mylocalpackage` is symbolic link to -> `..\mylocalpackage`

**Comment by basarat** [comment]
Got it .... perhaps `fs` is not able to read over symlinks. I'll investigate.

**Comment by zimme** [comment]
There's `fs.realpath`/`fs.realpathSync` which should resolve paths over symlinks too.

**Comment by basarat** [comment]
> after those commands otherpackage\node_modules\mylocalpackage is symbolic link to -> ..\mylocalpackage

@Ciantic And you have a reference between `mylocalpackage` and `otherpackage`. Can I have a few code files pasted here so I can test it. Sorry to raise this issue from the dead :rose:

**Comment by zimme** [comment]
I have a problem with symlinks too, don't know if it's the same issue as this.

I have a project with 3 microservices. Some code are used from all 3 microservices and I have setup `.d.ts` files for a few things in one of these, and I symlink to the definitions folder in that project.

These sub-projects have separate atom projects and because of that I don't get error squiggles in the project that have the plain `.d.ts` files and in the projects which have the symlinks to the folder I get error squiggles because this instance of typescript plugin doesn't understand following symlink for the definition files.

**Comment by basarat** [comment]
@zimme can you provide a two file repro. `a.ts` -ref-> `./b.ts` where `b.ts` is a symlink to `./foo/b.ts`. If that doesn't work ... I'll try and get a setup going here. Might be tsconfig (fileGlob) not working or `fs.exists` not working. Would be good to get a narrowed down repro

**Comment by zimme** [comment]
Go into `/foo` run `atom .` see error squiggles! Atom TS can't find `b.d.ts` with the definitions.

Symlinks work in general, but when a symlinked folder have folders with symlink inside them something breaks.

https://github.com/zimme/atom-typescript-symlink-error

**Comment by basarat** [comment]
@zimme git clone gave a very weird REPO : 

![image](https://cloud.githubusercontent.com/assets/874898/10567239/a316f784-7649-11e5-95df-4e68aa107933.png)

The symlinks came down as Files (same as what you see online https://github.com/zimme/atom-typescript-symlink-error/blob/master/definitions/test/b.d.ts)

**Comment by basarat** [comment]
Add a `filesGlob` to `tsconfig.json` and you see that b.d.ts doesn't get listed: 

```json
{
	"compileOnSave": false,
	"filesGlob": [
		"**/*.ts",
		"**/*.tsx",
		"!node_modules/**"
	],
	"compilerOptions": {},
	"files": [
		"a.ts"
	],
	"exclude": []
}
```

![image](https://cloud.githubusercontent.com/assets/874898/10567272/a6473b48-764a-11e5-89e9-18bf62768d51.png)

Marking as a bug for tsconfig. 

/cc @blakeembrey

**Comment by blakeembrey** [comment]
@basarat I created a branch here: https://github.com/TypeStrong/tsconfig/pull/13. Would you mind giving that a shot, and if I works I'll do a new release. I imagine that's all it takes, is enabling `follow: true` for symlinks.

**Comment by basarat** [comment]
@blakeembrey yup that fixes it

![image](https://cloud.githubusercontent.com/assets/874898/10569819/5d447abe-7677-11e5-8137-a89f3dab1ef3.png)


Publish whenever :rose:

**Comment by blakeembrey** [comment]
Done :+1: Release with `v2.0.1`.

**Comment by basarat** [comment]
@blakeembrey I'm doing it now :rose:

**Comment by basarat** [comment]
Fixed with `7.4.2` https://github.com/TypeStrong/atom-typescript/releases/tag/v7.4.2 :rose:

**Comment by zimme** [comment]
Thank you, much appreciated. Latest release works like a charm.

@basarat I'm on OSX, and I guess that's why stuff didn't work out of the box for you.
